### ISSUE TITLE
add back model/view_timezone options failed

### ISSUE THREAD

**Issue by bokonet** [symfony/symfony#13380]
I think the "add back model/view_timezone options" is not finished jet. 
Instead of "$options['model_timezone'], $options['view_timezone']" there is a lot of "null, null" in the code.

commit 1: https://github.com/symfony/symfony/commit/849fb2998485639d253e2438ecf78f62cc833a53

commit 2: https://github.com/symfony/symfony/commit/0104f15e5000efd3c0e9fc156812cc6ba946c5b2

commit 3: https://github.com/symfony/symfony/commit/1c4a75a00a86df3e17979be86bddaafc108daf98

Not sure, but commit 3 is correcting commit 1, but not commit 2.

**Comment by jakzal** [comment]
The third commit you listed by @xabbuh adds the two unfortunate options back. Options are brought back to avoid breaking existing apps. They're no point using them though.

What's the behaviour you're experiencing as opposed to what you expect?

**Comment by stof** [comment]
I think this actually also highlights a much bigger issue in the Form component (and also in Doctrine for that matter): using a PHP DateTime object to represent only a date or only a time is actually flawed. PHP does not know we care only about the date or only about the time. The object actually always represents a datetime, which is affected by timezones.
Unfortunately, PHP does not have a Date or a Time classes which could be used there

**Comment by bokonet** [comment]
I persist the formComponents "date" and "time" in SQL as DATE and TIME, so there is no timezone information needed. Stof is right, doctrine will translate me the date or time to an php \Datetime object, because php could not known/do better. I think this object will be created with the server default-timezone.

With commit 1 https://github.com/symfony/symfony/commit/849fb2998485639d253e2438ecf78f62cc833a53 my form looks like this:
`$builder->add('utcdate','date',array('required'=>FALSE));//is always UTC`
The form will give me back a datetime object with UTC timezone. Now i have the problem that for doctrine the entity has been modified, because of the timezone offset.

To fix this Problem, a add two new doctrine DBAL Types "UTCdateType" + "UTCtimeType" witch will translate the database values to an php \Datetime object with UTC-timezone. This solution worked perfekt for me: no more problems with timezones for time/date components and the server-default-timezone can be changed at any time.

With the changes of commit 2 https://github.com/symfony/symfony/commit/1c4a75a00a86df3e17979be86bddaafc108daf98 the form will create the php \Datetime object with the server default-timezone again and the timeoffset ist back again :(

Should be no problem, with commit 3 https://github.com/symfony/symfony/commit/1c4a75a00a86df3e17979be86bddaafc108daf98 i will use the option "view_timezone" to get it back in UTC:
`$builder->add('utcdate','date',array('required'=>FALSE,'view_timezone'=>'UTC'));`

But this did not work. I am new to symfony, so i am realy not sure. I think the options "model_timezone" and "view_timezone" are missing in f.e. line 60+61 in file https://github.com/symfony/symfony/blob/2.6/src/Symfony/Component/Form/Extension/Core/Type/DateType.php

Before commit 1 https://github.com/symfony/symfony/commit/849fb2998485639d253e2438ecf78f62cc833a53 the two options are present with `... $options['model_timezone'], $options['view_timezone'] ...` and now: https://github.com/symfony/symfony/commit/849fb2998485639d253e2438ecf78f62cc833a53 `... null,null, ...`

Hopefully, i am not totally wrong
...and sorry for the bad english ;)

**Comment by Axinet** [comment]
Without possibility to set view_timezone my DateTime object with timezone "Europe/Warsaw" show wrong date in form! 
Entity stores only date in database and upon retrieval time part is set to 00:00:00 - midnight and server/php_config timezone "Europe/Warsaw" eg.: 
````
  -date: DateTime {#1066 ▼
    +"date": "2014-10-19 00:00:00"
    +"timezone_type": 3
    +"timezone": "Europe/Warsaw"
````
But in form input's field ('widget' => 'single_text') I got **18/10/2014** - it's a one day shift (when displayed time part found the problem - 2 hours shift due to timezone mismatch: 18/10/2014 22:00:00)! 

As @stof commented it's a problem with no Date only object in php. Althought it looks like current DateType from symfony is usable only with servers configured to UTC and only workaround is to write custom TzDateType?!

**Comment by xabbuh** [comment]
Luckily this has been fixed in the meantime in #14256. So I am going to close here.
